### PR TITLE
Custom vault loading test map

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[general][vault]

## What this does
adds a test map with a corresponding config file at config/testvaults.txt that loads vaults from all lines in this file, if not commented out with a #.
if no lines are found or specified, all vaults are loaded on this map.
also adds a teleporter generator to this map and some defense items to make reaching these vaults easier in it.

## Why it's good
good quick way to test vaults on a map without booting into a main one, since the other test maps don't have mechanisms for this.

## Changelog
:cl:
 * rscadd: A new test map for vault coders has been added, test_vault.dmm. The file at config/vaulttest.txt can be used to specify which vaults show up in it.

### CODE REVIEW

File: config-example/testvaults.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# List of map elements to load for the map "test_vault.dmm", leave the file below this line blank to load all vaults.
+
+/datum/map_element/vault/clownroid
+/datum/map_element/vault/sokoban
+/datum/map_element/vault/black_site_prism
+/datum/map_element/dungeon/ninja_dojo
+/datum/map_element/dungeon/wizard_den
```
Comment (Kurfursten) [inline config-example/testvaults.txt]: Why not just make the default all vaults instead of this list?

File: config-example/testvaults.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# List of map elements to load for the map "test_vault.dmm", leave the file below this line blank to load all vaults.
```
Comment (Kurfursten) [inline config-example/testvaults.txt]: comment implies it loads only vaults if empty but it seems to load all map elements

Comment (SECBATON-GRIFFON) [reply config-example/testvaults.txt]: it calls the proc get_map_element_objects if empty, and the default subtypes it searches for is /datum/map_element/vault, so it does only load vaults.

File: config-example/testvaults.txt (hunk 1 above)
Comment (SECBATON-GRIFFON) [reply config-example/testvaults.txt]: proof of concept of the list filtering, if you want that to be the default i can just put #s in front of them to keep the examples

Comment (Kurfursten) [reply config-example/testvaults.txt]: I feel like the most intuitive case is to have all vaults spawn yeah

Comment (SECBATON-GRIFFON) [reply config-example/testvaults.txt]: done that now

**Pull request merged**